rules/test/lldb/lldb_test.py

```diff
@@ -0,0 +1,342 @@
+    status = ctx.GetCompletionStatus()
+    if status == -1:
+        return 1
+
+    # This should check the exit code here?
+    test_result = ctx.GetTestResult()
+    logger.info("Got result" + str(test_result))
+
+    test_spec_dict = None
+    with open(spec_path, 'r') as test_spec:
+        test_spec_dict = json.load(test_spec)
+
+    expected_value = test_spec_dict["expected_value"]
+    # Possibly use a python testing library or matcher here, but keep it simple
+    # for now
+    if test_result["expression_result"] != expected_value:
+        logger.error("Failed Expected", expected_value, "Got",
+                     test_result["expression_result"])
+        return 1
+
+    logger.debug("SUCCESS")
+    return 0
+
+
+def cleanup(ctx):
+    shutil.rmtree(ctx.test_root)
+
+
+def run_lldb_test(ipa_path, sdk, device, spec_path):
+    if not os.path.exists(ipa_path) or not ipa_path.endswith(".ipa"):
+        logger.error("Missing IPA / [ --app ] %s", ipa_path)
+        exit(1)
+
+    # Consider handling other IPAs here or types
+    ipa_name = os.path.basename(ipa_path)
+    app_name = ipa_name.replace(".ipa", "")
+
+    if not sdk:
+        logger.error("Missing SDK / [ --sdk ]")
+        exit(1)
+
+    if not device:
+        logger.error("Missing device / [ --device ] ")
+        exit(1)
+
+    if not os.path.exists(spec_path):
+        logger.error("Missing spec / [ --spec]", spec_path)
+        exit(1)
+
+    with open(spec_path, 'r') as test_spec:
+        test_spec_dict = json.load(test_spec)
+        cmd = test_spec_dict["breakpoint_cmd"]
+
+        # Do some level of validation here - mainly to help the users figure out to
+        # use this. Warning: for now there is no timeout if the breakpoint
+        # doesn't resolve - consider failing fast here with a better message.
+        # Regardless, it's a broken state. The failure mode is the users
+        # timeout..
+        if (cmd.startswith("br ") or cmd.startswith("breakpoint ")) == False:
+            logger.error(
+                "breakpoint_cmd needs some breakpoint set to run to %s ", spec_path)
+            exit(1)
+
+    exit_code = None
+    ctx = None
+    logger.info("Got app name %s", app_name)
+    try:
+        test_root = setup_test_root(spec_path)
+        ctx = TestContext(None, app_name, str(test_root))
+
+        # Main runloop - polls completion status
+        sim_thread = threading.Thread(
+            target=sim_thread_entry, args=(ctx, device, sdk, ipa_path))
+        debugger_thread = threading.Thread(
+            target=lldb_thread_entry, args=(ctx, 1))
+        sim_thread.start()
+        debugger_thread.start()
+        while ctx.GetCompletionStatus() == None:
+            logger.debug('Main thread...')
+            time.sleep(1)
+        sim_thread.join()
+        debugger_thread.join()
+        exit_code = emit_test_result(ctx, spec_path)
+    except Exception:
+        traceback.print_exc()
+    finally:
+        cleanup(ctx)
```

Comment:
@thiagohmcruz yeah this _can't_ terminate the simulator I don't think because of concurrency - say you have 3 LLDB tests on the same machine at 1 time..

Reply:
I expect that it'd leave open the simulator service, e.g. the app. But the app under test and debugger are terminated - is that what you're seeing?

Reply:
>I expect that it'd leave open the simulator service, e.g. the app

@jerrymarino Yep, this is exactly what I'm seeing and you're right we do want to keep that so we can run tests on the same machine. Nvm my comment!